packages/react/src/Footnotes/Footnotes.test.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {render, cleanup} from '@testing-library/react'
+import '@testing-library/jest-dom'
+
+import {axe, toHaveNoViolations} from 'jest-axe'
+
+import {Footnotes} from './Footnotes'
+import '../test-utils/mocks/match-media-mock'
+
+expect.extend(toHaveNoViolations)
+
+describe('Footnotes', () => {
+  const mockItemText = 'There are now 100 million developers around the world using GitHub.'
+
+  afterEach(() => {
+    cleanup()
+    jest.clearAllMocks()
+  })
+
+  it('renders the default footnotes correctly', async () => {
+    const items = ['Citation 1', 'Citation 2', 'Citation 3']
+
+    const {getByTestId, container} = render(
+      <Footnotes data-testid="footnotes">
+        {items.map((citation, index) => (
+          <Footnotes.Item data-testid={`item-${index}`} key={citation}>
+            {citation}
+          </Footnotes.Item>
+        ))}
+      </Footnotes>,
+    )
+
+    const results = await axe(container)
+    expect(results).toHaveNoViolations()
+
+    const rootEl = getByTestId('footnotes')
+
+    expect(rootEl).toBeInTheDocument()
+    expect(rootEl.tagName).toEqual('OL')
+
+    let index = 0
+    for (const item of items) {
+      const itemEl = getByTestId(`item-${index}`)
+      expect(itemEl).toHaveTextContent(item)
+      expect(itemEl).toBeInTheDocument()
+      expect(itemEl.tagName).toEqual('LI')
+      index++
+    }
+  })
```

Comment:
Just a suggestion, but you could get rid of test-ids completely here. Testing Library lists them as their [lowest-priority queries](https://testing-library.com/docs/queries/about/#priority) and as such they recommend avoiding them unless absolutely necessary

```tsx
  it('renders the default footnotes correctly', async () => {
    const items = ['Citation 1', 'Citation 2', 'Citation 3']

    const {getByRole, container} = render(
      <Footnotes>
        {items.map((citation, index) => (
          <Footnotes.Item data-testid={`item-${index}`} key={citation}>
            {citation}
          </Footnotes.Item>
        ))}
      </Footnotes>,
    )

    const results = await axe(container)
    expect(results).toHaveNoViolations()

    const listEl = getByRole('list')

    expect(listEl).toBeInTheDocument()
    expect(listEl.tagName).toEqual('OL')

    for (const item of items) {
      expect(getByRole('listitem', {name: item})).toBeInTheDocument()
    }
  })
```

Reply:
Thanks.

`expect(getByRole('listitem', {name: item})).toBeInTheDocument()`

☝️ doesn't work because of the way the content is nested inside another tag of the listitem. Relevant WCAG guidelines [here](https://www.w3.org/TR/wai-aria-1.2/#namecalculation). I've instead used the `textContent` property on the listitem to infer the correct value. IMO it's a bit more convoluted now, but happy to keep it like this. 

If you're wondering why the assertion against tagName is still there, it's because the `listitem` role can be manually assigned to a different tag, so this makes sure that doesn't happen/